src/Identity/Extensions.Core/src/AuthenticatorTokenProvider.cs

```diff
@@ -48,7 +48,7 @@ public virtual async Task<bool> ValidateAsync(string purpose, string token, User
         {
             var key = await manager.GetAuthenticatorKeyAsync(user);
             int code;
-            if (!int.TryParse(token, out code))
+            if (!int.TryParse(token, out code) || key == null)
```

Comment:
```suggestion
            if (key is null || !int.TryParse(token, out code))
```
As the null-check is cheaper than trying to parse the int?

Reply:
Sure